### DISCUSSION TITLE
Hello all!

### DISCUSSION THREAD

**Discussion by mindcrime** [jason-lang/jason#106] [General]
Hi Jason crew. I'm Phil and I just started learning AgentSpeak / Jason. I've read the "Programming MultiAgent Systems in AgentSpeak with Jason" book once through - reading it like a novel (eg, no exercises or coding or anything). And now I am going back through it and doing the exercises, writing code, modifying some of the provided examples and generally messing around and trying to get my head wrapped fully around AgentSpeak and Jason.

I expect I'll have a few questions as I work through this, so I'm glad to see this "Discussions" feature turned on. And hopefully it won't take me *too* long to get to a point where I can help others as well.

And if anybody is wondering what I plan to do with AgentSpeak and Jason... truth be told, it's mostly an exploratory thing right now. My interests lie in the direction of AGI and I could see a BDI loop like system being a fairly integral part of that. That said, I think LLM's / ANN's and other techniques could also be part of a useful / interesting system in a hybrid fashion. So a lot of my time will go into looking at how a BDI agent can integrate with other other "things": neural networks, inference engines for inductive, deductive, and abductive inference, rule induction, maybe case-based reasoning, etc. 

Cheers!

**Comment by jomifred** [comment]
Hi Phillip, welcome onboard of Jason! 

I will be happy to follow your progress on the exercises and discuss the content. 

Regarding your quest, to put together BDI and LMM/NN/… is certainly relevant and challenging. My feeling is that we can propose integration “solutions” at the technical level (e.g. perception using a NN), but a more comprehensive view is far more complex (as discussed by people  at NeuroSymbolic community)

all the best,

Jomi

**Reply by mindcrime** [reply]
Agreed. And I do spend a modest amount of my time reading/studying/listening to material from the Neuro-Symbolic community. I only interact at the fringes as I'm not really a member of formal academia, so I might not describe myself as *member* of that community per-se, but I do take a lot of inspiration from that body of work.

**Comment by rbordini** [comment]
Hi Phil, that's fantastic! Very interesting topics indeed. Are you working
on this for personal interest or as part of an academic course, if I may
ask?

**Reply by mindcrime** [reply]
Purely personal interest. I'm not associated with any academic institution, and I just research the things that interest me.

**Comment by rbordini** [comment]
Fantastic!
